pkg/bot/mattermost.go

```diff
@@ -135,10 +139,25 @@ func NewMattermost(ctx context.Context, log logrus.FieldLogger, commGroupName st
 		botMentionRegex: botMentionRegex,
 		renderer:        NewMattermostRenderer(),
 		userNamesForID:  map[string]string{},
+		messages:        make(chan mattermostMessage, 100),
+		messageWorkers:  pool.New().WithMaxGoroutines(10),
 	}, nil
 }
 
-// Start establishes mattermost connection and listens for messages
+func (b *Mattermost) startMessageProcessor(ctx context.Context) {
+	b.log.Info("Starting mattermost message processor...")
+	defer b.log.Info("Stopped mattermost message processor...")
+
+	for msg := range b.messages {
+		b.messageWorkers.Go(func() {
+			err := b.handleMessage(ctx, msg)
+			if err != nil {
+				wrappedErr := fmt.Errorf("while handling message: %w", err)
+				b.log.Errorf(wrappedErr.Error())
+			}
```

Comment:
let's unify that with other log messages:

```suggestion
			if err != nil {
				b.log.WithError(err).Error("Failed to handle Mattermost message")
			}
```